blogs/ecs-cross-account/account_backend/redis_and_database.yml

```diff
@@ -0,0 +1,331 @@
+Parameters:
+  ProjectName:
+    Type: String
+    Description: Project name to link stacks
+    Default: am-ecs-multi-account
+
+  EnvoyImage:
+    Type: String
+    Description: Envoy container image
+    Default: 840364872350.dkr.ecr.eu-central-1.amazonaws.com/aws-appmesh-envoy:v1.12.5.0-prod
+
+  RedisImage:
+    Type: String
+    Description: Container image for Redis
+    Default: redis:4.0.2
+
+  DatabaseImage:
+    Type: String
+    Description: Container image for Database
+    Default: mreferre/yelb-db:0.5
+
+  MeshName:
+    Type: String
+    Description: Name of the Mesh
+    Default: yelb
+
+Resources:
+  TaskSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: "Security group for the tasks"
+      VpcId:
+        Fn::ImportValue: !Sub "${ProjectName}:VPC"
+      SecurityGroupIngress:
+        - CidrIp:
+            Fn::ImportValue: !Sub "${ProjectName}:VpcCIDR"
+          IpProtocol: -1
+
+  LogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "${ProjectName}-log-group"
+      RetentionInDays: 30
+
+  TaskIamRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument: |
+        {
+          "Statement": [{
+            "Effect": "Allow",
+            "Principal": { "Service": [ "ecs-tasks.amazonaws.com" ]},
+            "Action": [ "sts:AssumeRole" ]
+          }]
+        }
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/CloudWatchFullAccess
+        - arn:aws:iam::aws:policy/AWSXRayDaemonWriteAccess
+        - arn:aws:iam::aws:policy/AWSAppMeshEnvoyAccess
+
+  TaskExecutionIamRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument: |
+        {
+          "Statement": [{
+            "Effect": "Allow",
+            "Principal": { "Service": [ "ecs-tasks.amazonaws.com" ]},
+            "Action": [ "sts:AssumeRole" ]
+          }]
+        }
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+        - arn:aws:iam::aws:policy/CloudWatchLogsFullAccess
+
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: "Security group for the instances"
+      VpcId:
+        Fn::ImportValue: !Sub "${ProjectName}:VPC"
+      SecurityGroupIngress:
+        - CidrIp:
+            Fn::ImportValue: !Sub "${ProjectName}:VpcCIDR"
+          IpProtocol: -1
+
+  DatabaseRegistry:
+    Type: AWS::ServiceDiscovery::Service
+    Properties:
+      Name: "yelb-db"
+      DnsConfig:
+        NamespaceId:
+          Fn::ImportValue: !Sub "${ProjectName}:ECSServiceDiscoveryNamespace"
+        DnsRecords:
+          - Type: A
+            TTL: 300
+      HealthCheckCustomConfig:
+        FailureThreshold: 1
+
+  DatabaseTaskDef:
```

Comment:
I think all of the TaskDef's need a dependsOn for the envoy container, so the service doesn't come up ahead of the proxy. e.g.

```
"dependsOn": [
        {
          "containerName": "envoy",
          "condition": "HEALTHY"
        }
      ]
```

Reply:
Added dependsOn for x-ray and envoy to every taskdef.